semgrep/semgrep/target_manager.py

```diff
@@ -209,10 +210,29 @@ def filter_includes(arr: Set[Path], includes: List[str]) -> Set[Path]:
     @staticmethod
     def filter_excludes(arr: Set[Path], excludes: List[str]) -> Set[Path]:
         """
-        Returns all elements in arr that do not match any excludes excludes
+        Returns all elements in arr that do not match any excludes pattern
         """
         return set(elem for elem in arr if not TargetManager.match_glob(elem, excludes))
 
+    @staticmethod
+    def filter_by_size(arr: Set[Path], max_target_size: int) -> Set[Path]:
+        """
+        Return all the files whose size doesn't exceed the limit.
+
+        If max_target_size is zero or negative, all paths are returned.
+        If some paths are invalid, they may or may not be included in the
+        result.
+        """
+        if max_target_size <= 0:
+            return arr
+        else:
+            return set(
+                path
+                for path in arr
+                if TargetManager._is_valid(path)
+                and os.path.getsize(path) <= max_target_size
+            )
```

Comment:
Suggestion:

```suggestion
            return {
                path
                for path in arr
                if TargetManager._is_valid(path)
                and os.path.getsize(path) <= max_target_size
            }
```

Reply:
instead of os.path.getsize we can also call path.stat().st_size but I think functionally both do the same thing